File: app/src/at-phuongle/java/com/asiantech/intern20summer1/intro/TheJourneyBegins.java
package com.asiantech.intern20summer1.intro;

public class TheJourneyBegins {
    /**
     * Write a function that returns the sum of two numbers.
     */
    int add(int param1, int param2) {
        return param1 + param2;
    }

    /**
     * Write a function that returns the century when input year
     */
    int centuryFromYear(int year) {
        if (year <= 100) {
            return 1;
        } else {
            return ((year) % 100 == 0) ? (year / 100) : ((year / 100) + 1);
        }
    }

    /**
     * Write a function that check the palindrome string
     */
    boolean checkPalindrome(String inputString) {
        int n = inputString.length();
        for (int i = 0; i < (n / 2); i++) {
            char a1 = inputString.charAt(i);
            char a2 = inputString.charAt(n - i - 1);

            if (a1 != a2) {
                return false;
            }
        }
        return true;
    }
}

File: app/src/at-sonnguyen/java/com/asiantech/intern20summer1/kotlin/TheJourneyBegins.kt
package com.asiantech.intern20summer1.kotlin

object TheJourneyBegins{
    /*
    *   returns the sum of two numbers.
    */
    private fun add(param1: Int, param2: Int) = param1+param2
    /*
    *   Given a year, return the century it is in.
    */
    private fun centuryFromYear(year: Int): Int = if(year%100 == 0 ) year/100 else year/100 +1
    /*
    *   Given the string, check if it is a palindrome.
    */
    private fun checkPalindrome(inputString: String): Boolean = (inputString.reversed()  == inputString)
    @JvmStatic
    fun main(args: Array<String>) {
        var num1 = 15
        var num2 = 30
        var years : Int = 2000
        var inputString : String = "abacaba"
        println("1:  $num1 + $num2 = ${add(num1,num2)}")
        println("2. $years in ${centuryFromYear(years)} century ")
        println("3. $inputString is a palindome ? ${checkPalindrome(inputString)}")
    }

}